Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-327.28.3.el7.x86_64)
 GCC:gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
Sha: 309ce893aa5803001ec3de9d61b53e736e6bca7f
Build: success
Build: success
Install HPCC Platform
Number of Error(s): 3
src/WsTopology.ts(55,57): error TS2339: Property \removeSpecialCharacters\ does not exist on type \typeof import(/mnt/disk1/home/vamosax/smoketest/PR-13025/build/esp/src/tmp/src/Utility)\.
CMake Error at esp/src/cmake_install.cmake:84 (message):
make: *** [Makefile:74: install] Error 1
CMake Error at esp/src/cmake_install.cmake:84 (message):
  npm run compile failed see above for details.
Call Stack (most recent call first):
  esp/cmake_install.cmake:46 (include)
  cmake_install.cmake:55 (include)
make: *** [Makefile:74: install] Error 1
Number of Error(s): 3
src/WsTopology.ts(55,57): error TS2339: Property \removeSpecialCharacters\ does not exist on type \typeof import(/mnt/disk1/home/vamosax/smoketest/PR-13025/build/esp/src/tmp/src/Utility)\.
CMake Error at esp/src/cmake_install.cmake:84 (message):
make: *** [Makefile:74: install] Error 1
CMakeFiles/CMakeOutput.log copied into ../CMakeOutput-2019-09-26_11-04-55.log
CMakeFiles/CMakeError.log copied into ../CMakeError-2019-09-26_11-04-55.log
Get ECLWatch build result
ECL Watch:  Rebuilding Site
---- Build Target: /mnt/disk1/home/vamosax/smoketest/PR-13025/build/esp/src/build
---- Creating out of source build environment
---- Installing third party libraries
---- Copy resources
---- Build website
src/WsTopology.ts(55,57): error TS2339: Property \removeSpecialCharacters\ does not exist on type \typeof import(/mnt/disk1/home/vamosax/smoketest/PR-13025/build/esp/src/tmp/src/Utility)\.
CMake Error at esp/src/cmake_install.cmake:84 (message):
  npm run compile failed see above for details.
Call Stack (most recent call first):
  esp/cmake_install.cmake:46 (include)
  cmake_install.cmake:55 (include)
make: *** [Makefile:74: install] Error 1
Number of Error(s): 3
src/WsTopology.ts(55,57): error TS2339: Property \removeSpecialCharacters\ does not exist on type \typeof import(/mnt/disk1/home/vamosax/smoketest/PR-13025/build/esp/src/tmp/src/Utility)\.
CMake Error at esp/src/cmake_install.cmake:84 (message):
make: *** [Makefile:74: install] Error 1
CMakeFiles/CMakeOutput.log copied into ../CMakeOutput-2019-09-26_11-04-55.log
CMakeFiles/CMakeError.log copied into ../CMakeError-2019-09-26_11-04-55.log

[Build error:](https://gist.github.com/ae88607d4e2fc783e7bfda7f19c6e304)
Build log [Build log](https://gist.github.com/HPCCSmoketest/ae88607d4e2fc783e7bfda7f19c6e304#file-result-19-09-26-11-04-25-log)

--- a/esp/src/eclwatch/LogWidget.js
+++ b/esp/src/eclwatch/LogWidget.js
@@ -197,13 +197,13 @@ define([
                     store: store,
                     query: this.getFilter(),
                     columns: {
-                        lineNo: {width: 80 },
-                        audience: { label: this.i18n.Audience, width: 60},
-                        date: { label: this.i18n.Date, width: 100 },
-                        time: { label: this.i18n.Time, width: 100 },
-                        pid: { label: "PID", width: 60 },
-                        tid: { label: "TID", width: 60 },
-                        details: { label: this.i18n.Details, width: 1024 },
+                        MsgID: {width: 80 },
+                        Audience: { label: this.i18n.Audience, width: 60},
+                        Date: { label: this.i18n.Date, width: 100 },
+                        Timemilli: { label: this.i18n.Time, width: 100 },
+                        PID: { label: "PID", width: 60 },
+                        TID: { label: "TID", width: 60 },
+                        Details: { label: this.i18n.Details, width: 1024 },
                         dummy: { label: "", width: 0 }
                     }
                 }, this.id + "LogGrid");

--- a/esp/src/src/Utility.ts
+++ b/esp/src/src/Utility.ts
@@ -126,6 +126,10 @@ export function valueCleanUp(intsize): string {
     }
 }
 
+export function removeSpecialCharacters(stringToConvert): string {
+    return stringToConvert.replace(/[\!\@\#\$\%\^\&\*\)\(\+\=\.\<\>\{\}\[\]\:\;\'\"\|\~\`\_\-]/g, "");
+}
+
 export function unitTest(size, unit) {
     var nsIndex = size.indexOf(unit);
     if (nsIndex !== -1) {

--- a/esp/src/src/WsTopology.ts
+++ b/esp/src/src/WsTopology.ts
@@ -8,6 +8,7 @@ import * as QueryResults from "dojo/store/util/QueryResults";
 import * as Evented from "dojo/Evented";
 
 import * as ESPRequest from "./ESPRequest";
+import * as Utility from "./Utility";
 
 declare const dojoConfig;
 
@@ -39,31 +40,30 @@ var TpLogFileStore = declare([Memory, Evented], {
             }).then(lang.hitch(this, function (response) {
                 var data = [];
                 if (lang.exists("TpLogFileResponse.LogData", response)) {
+                    var columns = response.TpLogFileResponse.LogFieldNames.Item;
                     this.lastPage = response.TpLogFileResponse.LogData;
                     this.emit("pageLoaded", this.lastPage);
                     arrayUtil.forEach(response.TpLogFileResponse.LogData.split("\n"), function (item, idx) {
                         if (options.start === 0 || idx > 0) {
                             //  Throw away first line as it will probably only be a partial line  ---
                             var itemParts = item.split(" ");
-                            var lineNo, audience, time, pid, tid, date, details;
-                            if (itemParts.length) lineNo = nextItem(itemParts);
-                            if (itemParts.length) audience = nextItem(itemParts);
-                            if (itemParts.length) date = nextItem(itemParts);
-                            if (itemParts.length) time = nextItem(itemParts);
-                            if (itemParts.length) pid = nextItem(itemParts);
-                            if (itemParts.length) tid = nextItem(itemParts);
-                            if (itemParts.length) details = itemParts.join(" ");
+                            var tempObj = {
+                                __hpcc_id: response.TpLogFileResponse.PageNumber + "_" + idx
+                            };
 
-                            data.push({
-                                __hpcc_id: response.TpLogFileResponse.PageNumber + "_" + idx,
-                                lineNo: lineNo,
-                                audience: audience,
-                                date: date,
-                                time: time,
-                                pid: pid,
-                                tid: tid,
-                                details: details
-                            });
+                            for  (var i = 0; i < columns.length; ++i) {
+                                var cleanName = Utility.removeSpecialCharacters(columns[i]);
+                                var value = "";
+
+                                if ((i + 1) == (columns).length) {
+                                    value = itemParts.join("");
+                                } else if (itemParts.length) {
+                                    nextItem(itemParts);
+                                }
+
+                                tempObj[cleanName] = value;
+                                data.push(tempObj);
+                            }
                         }
                     }, this);
                 }

